extensions/expath/src/org/expath/httpclient/model/exist/EXistResult.java

```diff
@@ -94,8 +92,13 @@ public void add(final Reader reader) throws HttpClientException {
     @Override
     public void add(final InputStream is) throws HttpClientException {
         try {
-            result.add(BinaryValueFromInputStream.getInstance(context, new Base64BinaryValueType(), is));
-        } catch(final XPathException xpe) {
+            // we have to make a temporary copy of the data stream, as the socket will be closed shortly
+            final TemporaryFileManager temporaryFileManager = TemporaryFileManager.getInstance();
+            final Path tempFile = temporaryFileManager.getTemporaryFile();
```

Comment:
Will this for small files, a in-memory operation? I ask for performance reasons.... (IO overhead to disk makes things slow)

Reply:
I don't think you will notice the difference. As Network is so much slower than disk in the first place